File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution(object):
    def multi(self,n):
        ans=1
        while(n>0):
            ans=ans*(n%10)
            n=n/10
        return ans

    def smallestNumber(self, n, t):
        while(n>0):
            d=self.multi(n)
            if d%t==0:
                return n
            else:
                 n=n+1
        return -1
        

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int multi(int n){
        int ans;
        int digit=1;
        while(n>0){
        digit=(digit*(n%10));
        n=n/10;
        }
        return digit;
    }
    int smallestNumber(int n, int t) {
        while(n>0){
          int d=multi(n);
         if(d%t==0)return n;
         else n++;
        }
        return -1;
    }
};